pandas/tests/series/indexing/test_setitem.py

```diff
@@ -570,6 +573,9 @@ def test_slice_key(self, obj, key, expected, val, indexer_sli, is_inplace):
         indkey = np.array(ilkey)
         self.check_indexer(obj, indkey, expected, val, indexer_sli, is_inplace)
 
+        genkey = (x for x in indkey)
+        self.check_indexer(obj, genkey, expected, val, indexer_sli, is_inplace)
+
```

Comment:
these tests are good, but only cover Series.  do we need analogous DataFrame tests?

Reply:
@phofl i think this was the main outstanding question.  if the answer is "no" then this should be good to go (merge master just to be on the safe side)

Reply:
Sorry for the long wait. I have more time on my handy in the coming weeks and months again :)

One-dimensional indexers on DataFrames work, but two dimensionals like

```
df = DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})

indexer = (x for x in [1, 2])
df.iloc[indexer]
```

still don't, because we access the tuple elements to check things before doing anything, which leaves an empty generator behind. Only way I am seeing currently is converting this directly in __setitem__ and __getitem__. Thoughts @jbrockmendel ?

Reply:
> Only way I am seeing currently is converting this directly in setitem and getitem

I expect this is probably right

Reply:
Put something up then tomorrow or the day after